core/templates/xregistry_openapi_template.json

```diff
@@ -1,1068 +1,1937 @@
+                        "schema": { "type": "integer" }
+                    },
+                    {
+                        "in": "query",
+                        "name": "top",
+                        "description": "The maximum number of versions to return.",
+                        "required": false,
+                        "schema": { "type": "integer" }
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The collection of versions.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "{%-resourceTypeReference-%}"
+                                }
+                            }
+                        }
+                    },
+                    "400": {
+                        "$ref": "#/components/responses/BadRequest"
+                    },
+                    "401": {
+                        "$ref": "#/components/responses/Unauthorized"
+                    },
+                    "404": {
+                        "$ref": "#/components/responses/NotFound"
+                    },
+                    "500": {
+                        "$ref": "#/components/responses/InternalServerError"
+                    }
+                }
+            },
+            "post": {
+                "operationId": "post{%-resourceNameSingular-%}Versions",
+                "tags": ["ResourceVersion"],
+                "description": "Creates or updates one or more versions of a resource. Query parameter 'setdefaultversionid' may be used to designate the new default version. (See spec: Creating or Updating Versions)",
+                "parameters": [
+                    {
+                        "in": "query",
+                        "name": "setdefaultversionid",
+                        "description": "Specifies the versionid for the default version. Use 'null' to reset to newest-default. (See spec: Default Version of a Resource)",
+                        "required": false,
+                        "schema": { "type": "string" }
+                    },
+                    {
+                        "$ref": "#/components/parameters/inline"
+                    },
+                    {
+                        "$ref": "#/components/parameters/filter"
+                    },
+                    {
+                        "$ref": "#/components/parameters/doc"
+                    },
+                    {
+                        "$ref": "#/components/parameters/specversion"
+                    }
+                ],
+                "requestBody": {
+                    "description": "A map of version definitions keyed by versionid.",
+                    "content": {
+                        "application/json": {
+                            "schema": {
+                                "type": "object",
```

Comment:
does openapi have the notion of a map? If so, it might be more appropriate here (and other POSTs too)